app/diagnostics.go

```diff
@@ -1110,3 +1061,128 @@ func (s *Server) trackChannelModeration() {
 		"use_channel_mentions_guest_disabled_count": useChannelMentionsGuest,
 	})
 }
+
+func (s *Server) trackPluginConfig(cfg *model.Config) {
+	pluginConfigData := map[string]interface{}{
+		"enable_nps_survey":             pluginSetting(&cfg.PluginSettings, "com.mattermost.nps", "enablesurvey", true),
+		"enable":                        *cfg.PluginSettings.Enable,
+		"enable_uploads":                *cfg.PluginSettings.EnableUploads,
+		"allow_insecure_download_url":   *cfg.PluginSettings.AllowInsecureDownloadUrl,
+		"enable_health_check":           *cfg.PluginSettings.EnableHealthCheck,
+		"enable_marketplace":            *cfg.PluginSettings.EnableMarketplace,
+		"require_pluginSignature":       *cfg.PluginSettings.RequirePluginSignature,
+		"enable_remote_marketplace":     *cfg.PluginSettings.EnableRemoteMarketplace,
+		"automatic_prepackaged_plugins": *cfg.PluginSettings.AutomaticPrepackagedPlugins,
+		"is_default_marketplace_url":    isDefault(*cfg.PluginSettings.MarketplaceUrl, model.PLUGIN_SETTINGS_DEFAULT_MARKETPLACE_URL),
+		"signature_public_key_files":    len(cfg.PluginSettings.SignaturePublicKeyFiles),
+	}
+
+	// knownPlugins maps plugin IDs to telemetry suffixes
+	var knownPlugins = map[string]string{
```

Comment:
Since we fully control the dataplane now, should we just sync with the team there and have them maintain the historical mapping?

Reply:
@Adovenmuehle do you have any thoughts on this?

Reply:
Would this mean that, for instance, instead of us getting an `enable_gitlab` flag we would instead get `enable_com.github.manland.mattermost-plugin-gitlab`? I'd prefer to keep the former if possible. I've also talked to @larkox and we were discussing having a plugin "short name" defined for each plugin so that we could just have `gitlab` instead of the `com.github...` one for use in telemetry.

Reply:
@Adovenmuehle, can you expand on the rationale for having "short" names sent by the application instead of the canonical ids? Is there a performance concern?

Reply:
Not performance, just usability on the Snowflake side. essentially each one of those names will become a column name in the database, so `com.mattermost.confluence` will end up as `ENABLE_COM_MATTERMOST_CONFLUENCE` which isn't necessarily horrible but it's just easier for it to be `ENABLE_CONFLUENCE`. Also, if we start adding more telemetry in the plugins themselves, and they standardize on the short name, it'll be a lot nicer to have track events prefixed by the short plugin name (e.g. `CONFLUENCE_SOME_EVENT` vs `COM_MATTERMOST_CONFLUENCE_SOME_EVENT`).

Reply:
There might be a future where multiple plugins of the same name exist in the Marketplace, e.g. an official JIRA plugin and a community written one. Using the plugin name as a id will cause problems in that case.

@Adovenmuehle Is it responsable to "prettify" the data  upstream e.g. in Snowflake?

Reply:
Yeah having collisions on short names would make it less than ideal.

I'm most invested in having the plugin specific telemetry (e.g. events that are sent directly from the plugin) to follow a nicer pattern than I am for these `enabled` flags.

For context, RudderStack is responsible for creating the column names in Snowflake (and we don't really have control over it).

Reply:
> I'm most invested in having the plugin specific telemetry (e.g. events that are sent directly from the plugin) to follow a nicer pattern than I am for these `enabled` flags.

@Adovenmuehle Plugin specific events are already using the plugin ID (https://github.com/mattermost/mattermost-plugin-api/blob/5b8d13c39e262600255d59e82d864db28ab4c1f1/experimental/telemetry/tracker.go#L80) Hence, I think it makes sense to also use the plugin ID for the `enabled` flags.

Reply:
I talked to @enelson720 and we're good to go for using the plugin IDs for the enabled flags and version numbers here. So you no longer need the `knownPlugins` map.